node/src/main/scala/coop/rchain/node/diagnostics/effects/package.scala

```diff
@@ -100,6 +100,9 @@ package object effects {
               }
         } yield r
 
+      override def trace[A](tag: String)(block: F[A])(implicit parentSource: Source): F[A] =
```

Comment:
PTAL at this new way to create traces, without need to always create explicit source for each trace.

Reply:
I've mentioned this in the first bullet on #3001. Macro to support this is already created. There are maybe some cases where you want to differentiate multiple traces inside one function (?).

Reply:
` There are maybe some cases where you want to differentiate multiple traces inside one function (?).` I'm not sure, I assume that if nested calls should give nested traces. So you create implicit parent source and then able to use this method to just provide tag one level down. But how to structure tracing in best way is a bit blurry for me.

Reply:
I'm thinking on multiple traces on the same level with the same parent source. Basically, source can be derived from the code itself in which case any manual definition can be avoided.

Reply:
I see, so we should start using this macro